### ISSUE TITLE
wezterm does not allow for ctrl+w+[hjkl] vim keys basically

### ISSUE THREAD

**Issue by r3k2** [wez/wezterm#6048]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

- Arch GNU/Linux
- Wayland
- sway

Hello decided to try out wezterm but for my surprise a lot of TUI programs that use VIM keys native like "iamb" for the matrix protocol do not work, I can't switch the TUI panel from room to room or do other VIM key combinations.. 
I did  tried to actually run vim (here they do work... so WTF? ) why for one program and not the other but same keys? 
I am just giving this a try because of sixel support, iamb also supports sixel and other TUI programs that I use.

### WezTerm version

20240203-110809-5046fc22

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Hello decided to try out wezterm but for my surprise a lot of TUI programs that use VIM keys native like "iamb" for the matrix protocol do not work, I can't switch the TUI panel from room to room or do other VIM key combinations.. 
I did  tried to actually run vim (here they do work... so WTF? ) why for one program and not the other but same keys? 
I am just giving this a try because of sixel support, iamb also supports sixel and other TUI programs that I use.

### To Reproduce

run a vim keys TUI program with wezterm, works on alacritty/toot/xterm etc fine.

### Configuration

on my ~/.config/   <--- standard for Linux not sure why wezterm defaults  to ~/ :shrug:  
below you see are just me trying with no success to dissable ctrl for weztern because I do not want nor need any weztern keybinds at all, (only maybe ctrl+t to open a new tab) everything else I use my TUI programs keybinds with vim keys.

```
local wezterm = require ("wezterm")
local config = wezterm.config_builder()

config.use_dead_keys = false
config.window_background_opacity = 0.8

config.color_scheme = "Batman"

config.font = wezterm.font("MesloLGS Nerd Font Mono")
config.font_size = 14

config.enable_tab_bar = false

-- config.window_decorations = "false"

config.window_background_opacity = 0.8
config.disable_default_key_bindings = true


config.keys = {
  {
    key = 'w',
    mods = 'CTRL',
    action = wezterm.action.DisableDefaultAssignment,
  },
}

return config
```

### Expected Behavior

to work like any other Linux terminal ever.

### Anything else?

in case I can't modify the ctrl+key+keys on wezterm I just rather dissable all keybinds for it and keep my original programs keybinds. 
thanks

**Comment by loops** [comment]
Hi r3k2,

What is the problem you're seeing?   Which specific key are you pressing, and getting the wrong behavior?

**Comment by r3k2** [comment]
it says above 4 times... read ? ctrl+w+[hjkl]  = vim keys anything ctrl+letter+letter just does not work on any of my 5 TUI that uses vim keys only on neovim works... I wrote already all this above

**Comment by loops** [comment]
Thanks.   Could you please describe what happens?  Does Wezterm generate the wrong characters, or nothing at all?  What keys do you press, and what does Wezterm actually generate at each stage?   

Since it works in Neovim, it sounds like this is some sort of configuration error.  Could you try running:

$  stty werase undef

And see if your TUI apps work?

**Comment by r3k2** [comment]
> Thanks. Could you please describe what happens? Does Wezterm generate the wrong characters, or nothing at all? What keys do you press, and what does Wezterm actually generate at each stage?

nothing absolutely nothing no error is like my program never gets  the request whatsoever. :( only happens with wezterm not sure why is different :thinking: 

> Since it works in Neovim, it sounds like this is some sort of configuration error. Could you try running:

well above is my current config but this started to happen since day 0 when I had no config whatsoever only defaults, then I started to play with the config and trying to "dissable" all weztern keybinds because honestly I dont need them.

> $ stty werase undef

I will try this on my work computer that is where I am testing weztern and update this
> And see if your TUI apps work?

the one I use most is IAMB https://github.com/ulyssa/iamb   none of the keybinds works specially the ones that need ctrl+something+something

**Comment by r3k2** [comment]
I run stty werase undef, same nothing happens.

**Comment by loops** [comment]
Well it's an odd problem for sure.   I just installed iamb on Fedora/Wayland/Sway, and it ran okay.  I was able to do a :split, and then use ctrl-wj and ctrl-wk  to move back and forth between the panes.  Was able to install the same wezterm version as yours:

[wezterm-20240203_110809_5046fc22-1.fedora39.x86_64.rpm](https://github.com/wez/wezterm/releases/download/20240203-110809-5046fc22/wezterm-20240203_110809_5046fc22-1.fedora39.x86_64.rpm)

So we need to dig into why wezterm is getting confused on your system.  I did try to run wezterm without any configuration at all, and that worked..  so you might want to try that as well:

wezterm --skip-config

**Comment by bew** [comment]
Try to run wezterm with:
`wezterm --skip-config --config debug_key_events=true`
([ref](https://wezfurlong.org/wezterm/config/lua/config/debug_key_events.html))
and give us the output when you press the keys you want

**Comment by r3k2** [comment]
ok im doing this now on my linux puseos tablet
- i was not able to open weztern and copy paste with ctrl+v or ctrl+shift+v to paste the command so had to open kterminal and paste it there and run weztern this way
- here is a screenshot of the output trying ctrl+w+h then with ctrl+w+l ctrl+w+k etc 
![iamb-wezterm](https://github.com/user-attachments/assets/b67786ce-3d14-4409-93f2-f851c282e104)

i have tried to pipe stdout to a file with no luck neither with tee

**Comment by bew** [comment]
This is not enough data from the debug, we barely see the ctrl-w being released..

If you can't use wezterm, open another terminal and start wezterm from there with the debug options, then **copy/paste** all the log lines in a block of code here in the issue, like this:
````
```
your
log lines
here
```
````

---

<details>
<summary>For example, here is my debug output for `Ctrl-w` then `h` then closing the wezterm window (on Linux, X11)</summary>

```
16:01:32.928  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(LeftControl), modifiers: NONE, leds: (empty), phys_code: Some(LeftControl), raw_code: 37, repeat_count: 1, key_is_down: true, handled: Handled(false) }
16:01:32.928  INFO   wezterm_gui::termwindow::keyevent > key_event KeyEvent { key: LeftControl, modifiers: NONE, leds: (empty), repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(LeftControl), modifiers: NONE, leds: (empty), phys_code: Some(LeftControl), raw_code: 37, repeat_count: 1, key_is_down: true, handled: Handled(false) }) }
16:01:32.928  INFO   wezterm_gui::termwindow::keyevent > send to pane DOWN key=LeftControl mods=NONE
16:01:33.355  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(Z), modifiers: CTRL, leds: (empty), phys_code: Some(Z), raw_code: 52, repeat_count: 1, key_is_down: true, handled: Handled(false) }
16:01:33.355  INFO   wezterm_gui::termwindow::keyevent > key_event KeyEvent { key: Char('w'), modifiers: CTRL, leds: (empty), repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(Z), modifiers: CTRL, leds: (empty), phys_code: Some(Z), raw_code: 52, repeat_count: 1, key_is_down: true, handled: Handled(false) }) }
16:01:33.355  INFO   wezterm_gui::termwindow::keyevent > send to pane DOWN key=Char('w') mods=CTRL
16:01:33.356  INFO   wezterm_term::terminalstate::keyboard > key_down: sending "\u{17}", Char('w') CTRL
16:01:33.454  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: Char('w'), modifiers: CTRL, leds: (empty), repeat_count: 1, key_is_down: false, raw: Some(RawKeyEvent { key: Physical(Z), modifiers: CTRL, leds: (empty), phys_code: Some(Z), raw_code: 52, repeat_count: 1, key_is_down: false, handled: Handled(false) }) }
16:01:33.454  INFO   wezterm_gui::termwindow::keyevent     > send to pane UP key=Char('w') mods=CTRL
16:01:36.092  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Physical(H), modifiers: CTRL, leds: (empty), phys_code: Some(H), raw_code: 43, repeat_count: 1, key_is_down: true, handled: Handled(false) }
16:01:36.092  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: Char('h'), modifiers: CTRL, leds: (empty), repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(H), modifiers: CTRL, leds: (empty), phys_code: Some(H), raw_code: 43, repeat_count: 1, key_is_down: true, handled: Handled(false) }) }
16:01:36.092  INFO   wezterm_gui::termwindow::keyevent     > send to pane DOWN key=Char('h') mods=CTRL
16:01:36.092  INFO   wezterm_term::terminalstate::keyboard > key_down: sending "\u{8}", Char('h') CTRL
16:01:36.169  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: Char('h'), modifiers: CTRL, leds: (empty), repeat_count: 1, key_is_down: false, raw: Some(RawKeyEvent { key: Physical(H), modifiers: CTRL, leds: (empty), phys_code: Some(H), raw_code: 43, repeat_count: 1, key_is_down: false, handled: Handled(false) }) }
16:01:36.169  INFO   wezterm_gui::termwindow::keyevent     > send to pane UP key=Char('h') mods=CTRL
16:01:36.946  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: LeftControl, modifiers: CTRL, leds: (empty), repeat_count: 1, key_is_down: false, raw: Some(RawKeyEvent { key: Physical(LeftControl), modifiers: CTRL, leds: (empty), phys_code: Some(LeftControl), raw_code: 37, repeat_count: 1, key_is_down: false, handled: Handled(false) }) }
16:01:36.946  INFO   wezterm_gui::termwindow::keyevent     > send to pane UP key=LeftControl mods=CTRL
```

</details>

**Comment by r3k2** [comment]
this is exactly what I did since weztern as I mention above did not let me run ctrl+v etc to paste your command so started it from Kterminal and try to pipe the stdout and to "tee" it and nothing  I am just going to give up and compile alacittry with sixel support this is taking to much effort when it works on any other terminal just fine, on real Linux not windows crap

**Comment by bew** [comment]
> try to pipe the stdout and to "tee" it and nothing

Log messages are sent to stdERR, not stdOUT ⚠️
So assuming you're using bash you can write `... &| pipe-to-this-command` or `... &> debug_output.txt`

**Comment by ulyssa** [comment]
@r3k2 have you updated to the latest iamb v0.0.10? I believe I fixed this issue. I was enabling xterm-style `modifyOtherKeys` but `crossterm` couldn't parse them.  You may also want to try enabling the Kitty enhanced key protocol in Wezterm (I believe it's off by default)

**Comment by r3k2** [comment]
Hello @ulyssa thanks for the reply, I am not sure if that will be the issue since I do not like kitty protocol was looking for a terminal with sixel support to use. Not just with IAMB but also other TUI clients that I use like RSS readers etc. that also employ vim keys, usually I have no problem with "foot" is what I usually use with Arch GNU/Linux,  but I got recently a Librem Linux tablet with pureOS based on Debian and was looking for a terminal that I could just add an apt repo and apt get install, is why I started to look at this project, but is Ok because I found out that I can also add a "foot" apt repo that is native to wayland and works great with Sixel image support, not sure why this terminal will break some of my TUI key binds  :man_shrugging:  with time I will have no problem trying to debug the issue but lately my time is minimal to many security breaches going on and clients asking for help. :hugs:
